src/main/scala/streamer/StreamerTop.scala

```diff
@@ -67,7 +68,39 @@ class StreamerTop(
   val streamer = Module(new Streamer(params, params.tagName))
 
   // io.csr and csrManager input connection
-  csr_manager.io.csr_config_in <> io.csr
+  csr_manager.io.csr_config_in.rsp <> io.csr.rsp
+
+  val csr_config_in_req_valid = WireInit(false.B)
+  val csr_config_in_req_bits = Wire(new CsrReq(csrAddrWidth))
+  val csr_config_in_req_ready = WireInit(false.B)
+
+  val streamerIdle2Busy = WireInit(false.B)
+  val streamerBusy2Idle = WireInit(false.B)
+
+  streamerIdle2Busy := streamer.io.busy_o && !RegNext(streamer.io.busy_o)
+  streamerBusy2Idle := !streamer.io.busy_o && RegNext(streamer.io.busy_o)
+
+  val performance_counter = RegInit(0.U(32.W))
+  when(streamer.io.busy_o) {
+    performance_counter := performance_counter + 1.U
+  }.elsewhen(streamerBusy2Idle) {
+    performance_counter := 0.U
+  }
```

Comment:
So this is to count how much time it took to process? Not how many cycles were successful yes?